library/src/main/java/com/datatorrent/lib/db/jdbc/AbstractJdbcPollInputOperator.java

```diff
@@ -438,119 +261,110 @@ public void endWindow()
     currentWindowRecoveryState = new MutablePair<>();
   }
 
-  public int getPartitionCount()
-  {
-    return partitionCount;
-  }
-
-  public void setPartitionCount(int partitionCount)
+  @Override
+  public void deactivate()
   {
-    this.partitionCount = partitionCount;
+    scanService.shutdownNow();
+    store.disconnect();
   }
 
-  @Override
-  public void activate(Context cntxt)
+  protected void pollRecords()
   {
-    if (context.getValue(OperatorContext.ACTIVATION_WINDOW_ID) != Stateless.WINDOW_ID
-        && context.getValue(OperatorContext.ACTIVATION_WINDOW_ID) < windowManager.getLargestRecoveryWindow()) {
-      // If it is a replay state, don't start any threads here
+    if (isPolled) {
       return;
     }
-  }
-
-  @Override
-  public void deactivate()
-  {
     try {
-      if (dbPoller != null && dbPoller.isAlive()) {
-        dbPoller.interrupt();
-        dbPoller.join();
+      ps.setFetchSize(getFetchSize());
+      ResultSet result = ps.executeQuery();
+      if (result.next()) {
+        do {
+          emitQueue.add(getTuple(result));
+        } while (result.next());
       }
-    } catch (InterruptedException ex) {
-      // log and ignore, ending execution anyway
-      LOG.error("exception in poller thread: ", ex);
+      isPolled = true;
```

Comment:
Does this mean that we want to poll records just once in a window?
Is this because the poll interval is sufficiently long so as to prevent multiple calls to ```pollRecords()``` within a single window?
Even in that case, if the poll interval is reduced, there might be multiple calls where only one call will happen in this case.